api/v1/intrusiondetection_types.go

```diff
@@ -17,11 +17,16 @@ limitations under the License.
 package v1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // IntrusionDetectionSpec defines the desired state of Tigera intrusion detection capabilities.
 type IntrusionDetectionSpec struct {
+	// ComponentResources can be used to customize the resource requirements for each component.
+	// Only DeepPacketInspection is supported for this spec.
+	// +optional
+	ComponentResources []IntrusionDetectionComponentResource `json:"componentResources,omitempty"`
```

Comment:
Do we support only 1 component for intrusion detection ?

Reply:
Only DPI supports customising resource requirements for now.

Reply:
Yes, but I assume we can have multiple DPI components in the future. Not just DeepPacketInspection.